model_read_only/security/ir.model.access.csv

```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+mrr1,mrr1,model_model_readonly_restriction,base.group_system,1,1,1,1
+mrr2,mrr2,model_model_readonly_restriction,base.group_user,1,0,0,0
```

Comment:
**suggestion(non-blocking)**: I wonder if the user should even have read access. An alternative could be to remove this permission, and access restrictions using `sudo()` when checking access rights.